lib/App/cpanminus/script.pm

```diff
@@ -553,126 +523,31 @@ sub numify_ver {
-        query => { nested => {
-            score_mode => 'max',
-            path => 'module',
-            query => { custom_score => {
-                metacpan_script => "score_version_numified",
-                query => { constant_score => {
-                    filter => { and => [
-                        { term => { 'module.authorized' => JSON::PP::true() } },
-                        { term => { 'module.indexed' => JSON::PP::true() } },
-                        { term => { 'module.name' => $module } },
-                        $self->version_to_query($module, $version),
-                    ] }
-                } },
-            } },
-        } },
-    } };
-
-    my $module_uri = "$metacpan_uri/file/_search?source=";
-    $module_uri .= $self->encode_json({
-        query => $query,
-        fields => [ 'date', 'release', 'author', 'module', 'status' ],
-    });
-
-    my($release, $author, $module_version);
+    my $metacpan_uri = 'https://fastapi.metacpan.org/v1/download_url/';
 
-    my $module_json = $self->get($module_uri);
-    my $module_meta = eval { JSON::PP::decode_json($module_json) };
-    my $match = $self->find_best_match($module_meta);
-    if ($match) {
-        $release = $match->{release};
-        $author = $match->{author};
-        my $module_matched = (grep { $_->{name} eq $module } @{$match->{module}})[0];
-        $module_version = $module_matched->{version};
-    }
+    my $url = $metacpan_uri . $module;
 
-    unless ($release) {
-        $self->chat("! Could not find a release matching $module ($version) on MetaCPAN.\n");
-        return;
-    }
-
-    my $dist_uri = "$metacpan_uri/release/_search?source=";
-    $dist_uri .= $self->encode_json({
-        filter => { and => [
-            { term => { 'release.name' => $release } },
-            { term => { 'release.author' => $author } },
-        ]},
-        fields => [ 'download_url', 'stat', 'status' ],
-    });
+    my $query = $self->uri_params(
+        ($version ? (version => $version) : ()),
+        ($dev_release ? (dev => 1) : ()),
+    );
+    $url .= '?' . $query
+        if length $query;
 
-    my $dist_json = $self->get($dist_uri);
-    my $dist_meta = eval { JSON::PP::decode_json($dist_json) };
+    my $dist_json = $self->get($url);
+    my $dist_meta = eval { $self->decode_json($dist_json) };
 
-    if ($dist_meta) {
-        $dist_meta = $dist_meta->{hits}{hits}[0]{fields};
-    }
     if ($dist_meta && $dist_meta->{download_url}) {
         (my $distfile = $dist_meta->{download_url}) =~ s!.+/authors/id/!!;
         local $self->{mirrors} = $self->{mirrors};
-        if ($dist_meta->{status} eq 'backpan') {
-            $self->{mirrors} = [ 'http://backpan.perl.org' ];
-        } elsif ($dist_meta->{stat}{mtime} > time()-24*60*60) {
-            $self->{mirrors} = [ 'http://cpan.metacpan.org' ];
-        }
-        return $self->cpan_module($module, $distfile, $module_version);
+        $self->{mirrors} = [ 'https://cpan.metacpan.org' ];
```

Comment:
Part of the reason for this is that cpan.metacpan.org is a backpan.  This routine is mainly used for complex version specifications, which will often need a backpan, and by --metacpan, which it makes sense to switch the mirror for.  I also don't really see a downside.  It could be refactored to handle this somewhat differently, but I'd rather that be addressed in a separate PR.